File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0, right = nums.length - 1;
        int mid = 0;
        while (left < right) {
            mid = (left + right) /2;

            if (nums[mid] < target) {
                left = mid + 1;
            } else if (nums[mid] > target) {
                right = mid - 1;
            } else {
                return mid;
            }

        }
        
        if (left == right && nums[left] < target) return left + 1;
        else return left; 
    }
}

File: 35-search-insert-position/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();

        int left = -1, right = n;

        while (right > left + 1) {
            int mid = (left + right) /2;
            if (nums[mid] < target) {
                left = mid;
            } else {
                right = mid;
            }
        }

    
        return right;
    }
};